### PR TITLE
Hw 6

### PR DESCRIPTION
**Pull request by AlenaKharytonchyk**
Router is installed. App loads.
A﻿ll items from desired behavior description above are implemented without defects.
End-to-end tests are updated to cover the logic above and they pass.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,9 +6,11 @@
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
+    "eslint-plugin-react-hooks": "^4.6.0",
     "focus-trap-react": "^10.1.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-router-dom": "^6.10.0",
```
Comment (AliaksandrZahorski) [inline package.json]: it is better to have constant number of library version

File: src/App.js [hunk 2]
```diff
@@ -1,46 +1,33 @@
...
     MovieDetails,
-    SortControl,
-    MovieForm
+    MovieListPage,
+    SearchForm
 } from "./components";
-import {moviesArray} from "./mockedMovies";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+
+export const routes = createBrowserRouter([
```
Comment (AliaksandrZahorski) [inline src/App.js]: Please, create new file for routes. And btw, why don't you use classic approach with `<Routers>`, I do not see any SSR in your task?

File: src/components/dialog/Dialog.scss [hunk 3]
```diff
@@ -1,5 +1,8 @@
 .modal-container {
-  position: relative;
+  position: absolute;
```
Comment (AliaksandrZahorski) [inline src/components/dialog/Dialog.scss]: is it the only one possible way to center element?

File: src/components/footer/Footer.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import './Footer.scss';
+import {Logo} from "../index";
+
+const Footer = () => {
+    return (
```
Comment (AliaksandrZahorski) [inline src/components/footer/Footer.js]: return is obsolete here

File: src/components/index.js [hunk 5]
```diff
@@ -11,6 +11,10 @@ import MovieDetails from "./movieDetails/MovieDetails";
 import SortControl from "./sortControl/SortControl";
 import Dialog from "./dialog/Dialog";
 import MovieForm from "./movieForm/MovieForm";
+import MovieListPage from "../containers/movieListPage/MovieListPage";
+import Footer from "./footer/Footer";
+import Logo from "./logo/Logo";
+import SearchForm from "./searchForm/SearchForm";
```
Comment (AliaksandrZahorski) [inline src/components/index.js:17]: I'd prefer see here and above something like import { SearchForm } from "./searchForm";

File: src/components/logo/Logo.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import './Logo.scss';
+
+const Logo = () => {
+    return (
```
Comment (AliaksandrZahorski) [inline src/components/logo/Logo.js]: obsolete return

File: src/components/movieDetails/MovieDetails.js [hunk 7]
```diff
@@ -1,40 +1,71 @@
...
+            const response = await fetch(`http://localhost:4000/movies/${movieId}`, {
+                signal: signal,
+            })
+            const data = await response.json();
+            console.warn("data", data)
+            controller = null;
+            setMovie(data);
+        }
+
+        fetchData();
```
Comment (AliaksandrZahorski) [inline src/components/movieDetails/MovieDetails.js]: fetchData(movieId); and you can move the rest of method code from component to utils

File: src/components/movieDetails/MovieDetails.js [hunk 8]
```diff
@@ -1,40 +1,71 @@
...
+    useEffect(() => {
+        let controller, signal;
+        const fetchData = async () => {
+            if (controller) {
+                controller.abort()
+            }
+            controller = new AbortController();
+            signal = controller.signal;
+
+            const response = await fetch(`http://localhost:4000/movies/${movieId}`, {
```
Comment (AliaksandrZahorski) [inline src/components/movieDetails/MovieDetails.js]: it is better to have constant.js file with export const BASE_URL = "http://localhost:4000";

File: package.json (hunk 1 above)
Comment (AlenaKharytonchyk) [reply package.json]: Done

File: src/App.js (hunk 2 above)
Comment (AlenaKharytonchyk) [reply src/App.js]: Created separate file.
I used the approach from documentation

File: src/components/dialog/Dialog.scss (hunk 3 above)
Comment (AlenaKharytonchyk) [reply src/components/dialog/Dialog.scss]: For sure not the only possible)

File: src/components/footer/Footer.js (hunk 4 above)
Comment (AlenaKharytonchyk) [reply src/components/footer/Footer.js]: Agree, thank you

File: src/components/movieDetails/MovieDetails.js (hunk 8 above)
Comment (AlenaKharytonchyk) [reply src/components/movieDetails/MovieDetails.js]: Thank you. Done

File: src/components/index.js (hunk 5 above)
Comment (AlenaKharytonchyk) [reply src/components/index.js:17]: I just import them here to later on import all components from one place like ../components

File: src/components/movieDetails/MovieDetails.js (hunk 7 above)
Comment (AlenaKharytonchyk) [reply src/components/movieDetails/MovieDetails.js]: Done

**Comment by AliaksandrZahorski** [comment]
approved

**Pull request merged**